src/cesride.udl

```diff
@@ -0,0 +1,322 @@
+[Error]
+enum Error {
+    "General",
+    "Matter",
+    "EmptyMaterial",
+    "Decode",
+    "UnexpectedCode",
+    "UnexpectedCountCode",
+    "UnexpectedOpCode",
+    "InvalidVarSize",
+    "InvalidVarRawSize",
+    "InvalidVarIndex",
+    "InvalidCodeSize",
+    "InvalidBase64Character",
+    "InvalidBase64Index",
+    "Shortage",
+    "EmptyQb64",
+    "UnknownSizage",
+    "UnknownHardage",
+    "UnknownBardage",
+    "UnsupportedSize",
+    "TooSmall",
+    "Prepad",
+    "NonZeroedPrepad",
+    "NonZeroedLeadByte",
+    "NonZeroedLeadBytes",
+    "NonZeroedPadBits",
+    "Parsing",
+    "ParseQb64",
+    "ParseQb2",
+    "Conversion",
+    "Value",
+    "Validation",
+    "Derivation"
+};
+
+enum CesrideMatterCodex {
+    "Ed25519Seed",
+    "Ed25519N",
+    "X25519",
+    "Ed25519",
+    "Blake3_256",
+    "Blake2b_256",
+    "Blake2s_256",
+    "SHA3_256",
+    "SHA2_256",
+    "ECDSA_256k1_Seed",
+    "Ed448_Seed",
+    "X448",
+    "Short",
+    "Big",
+    "X25519_Private",
+    "X25519_Cipher_Seed",
+    "Salt_128",
+    "Ed25519_Sig",
+    "ECDSA_256k1_Sig",
+    "Blake3_512",
+    "Blake2b_512",
+    "SHA3_512",
+    "SHA2_512",
+    "Long",
+    "ECDSA_256k1N",
+    "ECDSA_256k1",
+    "Ed448N",
+    "Ed448",
+    "Ed448_Sig",
+    "Tern",
+    "DateTime",
+    "X25519_Cipher_Salt",
+    "TBD1",
+    "TBD2",
+    "StrB64_L0",
+    "StrB64_L1",
+    "StrB64_L2",
+    "StrB64_Big_L0",
+    "StrB64_Big_L1",
+    "StrB64_Big_L2",
+    "Bytes_L0",
+    "Bytes_L1",
+    "Bytes_L2",
+    "Bytes_Big_L0",
+    "Bytes_Big_L1",
+    "Bytes_Big_L2",
+};
+
+enum CesrideIndexerCodex {
+    "Ed25519",
+    "Ed25519_Crt",
+    "ECDSA_256k1",
+    "ECDSA_256k1_Crt",
+    "Ed448",
+    "Ed448_Crt",
+    "Ed25519_Big",
+    "Ed25519_Big_Crt",
+    "ECDSA_256k1_Big",
+    "ECDSA_256k1_Big_Crt",
+    "Ed448_Big",
+    "Ed448_Big_Crt",
+    "TBD0",
+    "TBD1",
+    "TBD4",
+};
+
+dictionary Verfer {
+  sequence<u8> raw;
+  string code;
+  u32 size;
+};
+
+dictionary Prefixer {
+  sequence<u8> raw;
+  string code;
+  u32 size;
+};
+
+dictionary Dater {
+  sequence<u8> raw;
+  string code;
+  u32 size;
+};
+
+dictionary Seqner {
+  sequence<u8> raw;
+  string code;
+  u32 size;
+};
+
+dictionary Diger {
+  sequence<u8> raw;
+  string code;
+  u32 size;
+};
+
+dictionary Saider {
+  sequence<u8> raw;
+  string code;
+  u32 size;
+};
+
+dictionary Cigar {
+  sequence<u8> raw;
+  string code;
+  u32 size;
+  Verfer verfer;
+};
+
+dictionary Signer {
+  sequence<u8> raw;
+  string code;
+  u32 size;
+  Verfer verfer;
+};
+
+dictionary Siger {
+  sequence<u8> raw;
+  string code;
+  u32 index;
+  u32 ondex;
+  Verfer verfer;
+};
+
+namespace cesride {
+    [Throws=Error]
+    Dater dater_new_with_code_and_raw([ByRef] CesrideMatterCodex code, [ByRef] sequence<u8> raw);
```

Comment:
The new convenience methods make code optional in the concrete classes and are named `new_with_raw()`.. I think we should update that if we merge this PR

Reply:
done